simpsons/preprocess.py

```diff
@@ -0,0 +1,87 @@
+import csv
+import os
+
+import tensorflow as tf
+import sys
+
+
+def files_gen(annotations_path: str):
+    with open(annotations_path, 'rt') as csvfile:
+        spamreader = csv.reader(csvfile, delimiter=',')
+        for row in spamreader:
+            relative_path = row[0]
+            csv_base_path = os.path.dirname(annotations_path)
+            image_path = os.path.abspath(os.path.join(csv_base_path, relative_path))
+            ret = (image_path,
+                   int(row[1]), int(row[2]),  # upper left x,y
+                   int(row[3]), int(row[4]),  # lower right x,y
+                   row[5])
+            print(ret)
+            yield ret
+
+
+def read_crop_resize(path, x1, y1, x2, y2, label):
+    image_data = tf.read_file(path)
+
+    decoded = tf.image.decode_jpeg(image_data)
+
+    float32_img = tf.image.convert_image_dtype(decoded, tf.float32, saturate=True)
+
+    batched = tf.expand_dims(float32_img, axis=0)
+
+    cropped = tf.image.crop_to_bounding_box(
+        batched,
+        y1,
+        x1,
+        tf.maximum(1, y2 - y1),
+        tf.maximum(1, x2 - x1)
+    )
+
+    resized = tf.image.resize_bilinear(cropped, (300, 200))
+
+    # TODO Somewhere around here we need to turn the label into a 1-hot vec
+
+    return resized, label
+
+
+def encode_and_save(image, filename):
+    image_int = tf.image.convert_image_dtype(image[0], tf.uint8, saturate=True)
+    encoded_jpg = tf.image.encode_jpeg(image_int)
+    return tf.write_file(filename, encoded_jpg)
+
+
+def process(annotations_path, output_path):
+    # Create a dataset of filenames, bounding boxes, and string labels
+    filenames = tf.data.Dataset.from_generator(
+        lambda: files_gen(annotations_path),
```

Comment:
it looks like instead of using a nightly, we can simply use the workaround which is to convert the strings into bytes. would be better than having 2 tensorflow instances installed